tests/integration_tests/api/secrets_engines/test_aws.py

```diff
@@ -254,11 +254,16 @@ def test_read_role(self, label, configure_first=True, raises=None, exception_mes
                     d1=json.loads(read_role_response['data']['policy']),
                     d2=self.TEST_POLICY_DOCUMENT,
                 )
-            else:
+            elif 'credential_types' in read_role_response['data']:
```

Comment:
I don't have any issue with working around the `credential_type` / `credential_types` change in this secrets engine in this manner, but another option would be to go off the Vault version being tested against. E.g.:

```python
key_name = 'credential_types' if vault_version_lt('1.0.2') else 'credential_type'
self.assertEqual(
    first=read_role_response['data'][key_name],
    second=['iam_user'],
)
```

I like that route because it makes things a bit more explicit. What do you think?

Reply:
Sure! Changed and added a link to the corresponding vault commit.